### DISCUSSION TITLE
New ideas

### DISCUSSION THREAD

**Discussion by GoldyGuardian** [littletijn/spooky-view#17] [Ideas]
Spooky view is very cool! My pc looks really clean now. However it has some limitations. First of all:

Make a seperate setting for full screens. When opening an app in full screen, i would prefer not having it see trough. Or make it so that when an app is maximized, then all the background apps will get 0 opacity, THis makes having multiple apps open not annoying to see both of them at the same time.

Allow the user to easily tweak more things. For instance, i want images to be not see trough, but not for the rest. This makes using browsers while having it see trough much better. Or affect things that are not images and are dark/light, depedning on the settings. I have dark mode enabled, and then reading and looking at things would be much more easier, while still having a clean and very nice background.

Also it would be nice if the bug was fixed for when closing it, that the apps don't go back to full opacity.

English is not my first language, so if i made any spelling mistake then i am very sorry for that,

Thank you for this amazing programm!

**Comment by littletijn** [comment]
Thank you for using my program!

About that separate setting for full screen, you mean when a app window is maximized? I will have a look at it if it is possible to detect this state of a window.

The second request for only making a part transparent, it is quite limited. Only so called child windows can be made transparent in Windows 8.0 and newer. For browsers, it is not possible to only make images opaque while the rest is transparent. This is because the whole webpage is a single child window.

Can you give some examples of apps that will not return to full opaque when Spooky View is closed?

**Reply by GoldyGuardian** [reply]
My bad for not explaining correctly, i will do again for more clarification.

this is what i mean with the following terms:
Fullscreen: when you press F11, the app/programm becomes full screen. (or when you fullscreen a video that you are watching trough a browser)
maximized: is when you press on the square at the top right of any programm/app, It makes the smaller app larger to fit the current size of the screen.


Imagine the following scenarios:

If there is a programm maximized, make that x% opaque and the rest of the maximized programms should also be fully transparent, The programm that is the most if front and maximized should always be in opaque (these values should be able to be changed) This makes it so that other apps aren't visible behind the currently focused screen, which there is no point in having (or make that changable as a setting)
If there is a app/programm that becomes fullscreen, make that always 100%opaque (which the programm should detect when that changes)
If there is a maximized and a minimized app open, make them both opaque, so that when i focus on the smaller programm, the maximized programm doesn't become transparant. if there is also a way to keep the minimized app above the maximalized app, then that would also be really nice. That way, whenclicking on the maximalized app, there rest won't dissapear

if you make it so there are settings of these scenario, then that would be really apreciated!

if you want to replicate the scenarios, here is the settings i use:
![image](https://github.com/user-attachments/assets/377bde57-b71d-4dea-b93a-64a5c67445b3)
Both windows and app have ths exact setting.


a app/programm stays transparent when you have a fullscreen app open and closing spooky view, then it stays with the current transparency that you have applied. just press F11, And close spooky view, and it should stay transparent. (for me)


I also found out another bug,
When making a sreenshot using shift + windows + s, the apps/programms become more transparent. I tihnk this might happen because it applies the opacity decrease twice. (nm fixed it by disabling the opacity in the snipping tool)

Also thanks for the explenation about setting only a part transparent, that makes a lot of sence actually, unfortunately there are limitations like this. It would be really nice if windows would remove them. (which they probably won't)

Thank you again for making this programm, this is super easy to use, and with more customizations, this app would be really nice! I have been searching around the web for a while, and this is by far the best solution i found transparant trough apps.

Again, english isn't my first language, so i am sorry for any spelling mistakes.

**Reply by GoldyGuardian** [reply]
@littletijn ?

**Comment by littletijn** [comment]
Hello @GoldyGuardian! I just have noticed your answer. Sorry for the late reply.

If I implement the following settings, I guess I can make it that your scenarios are possible:
- Add two sliders for the opacity level for foreground and background maximized apps
- Add a option to make a window always on top

About that full screen with F11, I have to check if it is possible to detect this reliable. Because it might be implemented in a way that is hard to detect.

And that bug that a window will stay transparent when full screen, I just have reproduced it! Good find :) How I tested it:
- Set Spooky View to set transparency for every app
- Open Microsoft Egde
- Press F11 for full screen
- Open the Start Menu with the Windows button
- Close Spooky View that is now accessible by the visible task bar
- Microsoft Edge will still be transparent

A workaround is to first disable full screen and then close Spooky View

EDIT:
I have found out why a full screen window does not return opaque when closing Spooky View. It does not detect that the window could be modified by Spooky View. I have to figure out a way to make this work.
Also, the maximized state can be detected! So implementing this should be possible.

**Reply by GoldyGuardian** [reply]
Thank you for taking your time, to read and try to fix, and try to implement these ideas. That is very much appreciated!

i'll add a lot of things myself after i am done with my computing science bachelor.

**Comment by littletijn** [comment]
Hello @GoldyGuardian! I have made a build of the app with the bug that apps stay transparent when closing Spooky View fixed.
It should now return all apps back when closing.
It is attached to this post.
[SpookyView_1.0.7_Test1_x64_Signed_Portable.zip](https://github.com/user-attachments/files/17907932/SpookyView_1.0.7_Test1_x64_Signed_Portable.zip)

Could you check if this build does fix the problem when closing Spooky View for you?

I have checked if always on top is possible. I will try to add this option as well in a later build.

**Reply by GoldyGuardian** [reply]
Hello @littletijn !
I have downloaded your programm and ran it. It now works!

If you would add those 2 proeprties, that would be appreciated!

Also don't remember for the slider to also go from 0 to 100 instead of the 10 to 100!

As a fellow developer, i really appreaciate it. Not sure how long this took you to add, but probably some time. Thank you for hearing my advice and improving spooky view! I really like the app as its litterly the only thing i could find on the internet that does this. Its unfortunate that windows has annoying properties, making some things impossible to implement.

**Comment by littletijn** [comment]
I have the always on top option working. Only applying it when a new window is show is not working right now. Will try to fix that.

Also, I see so called IME windows. I guess I should not touch those, or at least check what those exactly are. They are most likely some window for text input in Japanese and other languages with complex scripts.

About that full screen, I haven't found a reliable way to detect that, because it are just windows that have been stretched. Maybe I can find some unique property to identify them.

Maximised windows are detectable, so that feature of transparency differences for those is possible. When I have time, I will try to release a new test build with this included.

**Reply by GoldyGuardian** [reply]
You are amazing! Thanks! GL with coding!

**Comment by littletijn** [comment]
Hello @GoldyGuardian!

Here is a third test build, now with the Always on top option! (The second one had the option missing in English)
[SpookyView_1.0.7_Test3_x64_Signed_Portable.zip](https://github.com/user-attachments/files/17968755/SpookyView_1.0.7_Test3_x64_Signed_Portable.zip)


A option for maximized windows is in the works. Fullscreen apps are not easy to detect and haven't found a solution yet.

**Reply by GoldyGuardian** [reply]
I have downloaded the test build, but either i am doing something wrong, or it does not work. Would you like to give me a little tutorial on what you should do to see its effects?

**Reply by littletijn** [reply]
There should be a option with "Always on top" in the setup window. This option is visible when a app is added or selected in the apps list. When enabled and applied, the windows of the selected app should now always be on top.

**Reply by GoldyGuardian** [reply]
For me, when applygin the always on top option, it immediately will apply it on top. But afterwards, when i click onm another programm, it will still move to the background.

It seems to only work when having spooky view as an active winodw.

PS: would you still like to change the limit from 10-100 to 0-100 so that we can also make things fully transparant when its not active?

**Reply by littletijn** [reply]
Hello @GoldyGuardian!
I have tried the Always on top option with Notepad++, Mozilla Firefox and the Windows 11 Settings app. Those apps the Always on top did apply as soon as set. Only with the Settings app when it was not minimized. With Microsoft Visual Studio 2022 it did only apply when set applying the settings two times.

About the transparency to 0, I will add it in the next build. When ready I will post it here again :)

Can you say with what app the Always on top option did not function? It might be a case that with that specific app it will not apply.

EDIT: Whoopsie! Found out, Always on top only works when transparency is also set. Let me fix that first ;)

**Reply by littletijn** [reply]
Hello again @GoldyGuardian!

Here is a new test build with the following changes:
- Transparency can be set to 0
- Always on top is now always correctly set when starting a app
- Always on top will work even when transparency is not set on a app

The new build is here:
[SpookyView_1.0.7_Test4_x64_Signed_Portable.zip](https://github.com/user-attachments/files/18129481/SpookyView_1.0.7_Test4_x64_Signed_Portable.zip)

If everything is working as is should now, I will make a new release with these changes! :)

**Reply by GoldyGuardian** [reply]
Hello @littletijn !

The test build is working almost perfectly.  Just 2 more feature and this app will be completed in my opinion. What i miss, is the main point of all of these other fetures, It is to have apps keep their active transparency even tough they are not active. if we have an minimalized which is always on top above a maximized screen, and you will focus on the always on top app, even if my other app which is maximalized is not focussed, i want it to keep its transparency for active apps. This property should be called "keep active transparency when maximized" Make this a boolean value. When set to true, it should keep its transparency, even when its not focussed, but only when its maximized.

Another problem this would probably create, is that there would be multiple maximized apps, which all have their transparency set to max because they are maximized. If there is a way to check the "op layer" of an maximized app, then only make that one the actie transparency.
Or just store a value which is the last active maximized app, and keep that one an active transparency. (Thats how i would do it, but i don't have any coding experience with Operating Systems.)

The other feature, is to be able to layer which programms are above eachother when set to always on top.

Thank you for adding the features previousously mentioned. Please don't try to rush this or feel obligated to add these features. If you don't feel like updating this app, then thats fine too. Either way good luck with coding!

If i don't explai nsomething clear enough, please ask me again!

**Comment by littletijn** [comment]
Hello @GoldyGuardian!

Was quite a busy week for me. I took some time to create the dialog for setting the maximized values. My plan was to implement it as in the following screenshot:

![image](https://github.com/user-attachments/assets/b4bc2c1c-367e-491b-bfdb-f1ae41c49bf4)

This way, the maximized state can be customized to any value desired. I think that this will cover your requests :)
I will try to add it next week.

**Reply by GoldyGuardian** [reply]
Thats crazy, thanks. Even tough only i am asking, you are making it for me! This app should get more users. I don't really know any other apps then this one that sets transparency.

**Comment by littletijn** [comment]
Hello @GoldyGuardian!

I had some time today, so here is a new build with the maximized option added! :)

[SpookyView_1.0.7_Test5_x64_Signed_Portable.zip](https://github.com/user-attachments/files/18270071/SpookyView_1.0.7_Test5_x64_Signed_Portable.zip)

For full-screen, first maximized the app, then set it to full-screen.

Hope that this will satisfy your request ;)

**Reply by GoldyGuardian** [reply]
I just downloaded your test build, and the extra settings are reallly nice to mess around with!

But, the main feature i wanted is to basically have the most front maximized app to have a transparency, and others not, even if you are not focussed. This is usefull because i only want one maximized windows to be opauge. So if you add a a property  (or 2) inside seperate property for maximized windows called: Front  maximized window transparency (and back maximized window transparency).

This should be the last feature i will request (for now).

If you add these, then it will be an app (iin my opinion) which can litterly do everything you need for transparant apps! this makes my windows look so nicer when using an animated background. Really hate how windows is not so easily customizable.

Thank you for adding those features.

**Reply by littletijn** [reply]
Hello @GoldyGuardian!

I have to look if it is possible to determine the current most top window in a specific state like maximized. Might need some additional changes. When I have some time again, I will check it out.

**Reply by GoldyGuardian** [reply]
Thanks, and this will also fix a bug where if you have 2 monitors, that the other monitor will always show the inactive value, even tough i still want it to be active.

**Comment by littletijn** [comment]
Hello @GoldyGuardian!

There is now a new pre-release available with some new features! Although it is missing the option for maximized windows, there is now support for hotkeys instead, making changes much easier to apply. There where some problems with maximized windows detection that I could not get right at the moment, so that is why they are missing in this version.

Download it from the following URL:
https://github.com/littletijn/spooky-view/releases/tag/v1.1.0-rc.1

Hopefully it might be useful for you :)

**Reply by GoldyGuardian** [reply]
Wow, amazing, i tought you forgot, but i was wrong. Very cool to see this get updated!! Thank you for updating this app on my request. I will give any feedback later!

**Reply by littletijn** [reply]
Just wanted to share with you that a new version is released! This is version 1.1.0.0.

You can download this version from Github at:
https://github.com/littletijn/spooky-view/releases/tag/v1.1.0

Or the Microsoft Store at:
https://apps.microsoft.com/detail/9PB88ZKT0CDB

A signed version will be released at a later day.
